.github/workflows/run-python-tests.yml

```diff
@@ -46,47 +43,38 @@ jobs:
     env:
       COVERAGE_REPORT: ${{ inputs.coverageReport }}
     steps:
-      - name: Checkout Code 🛎
-        uses: actions/checkout@v3
-      - name: Setup Conda Env 🐍
-        uses: conda-incubator/setup-miniconda@v2
+      - uses: actions/checkout@v4
+
+      # Cannot reuse `install-python-dependencies.yml` workflow as workflows cannot be called as steps, just as jobs
+      - uses: actions/setup-python@v4
         with:
-          miniforge-variant: Mambaforge
-          miniforge-version: latest
-          use-mamba: true
-          activate-environment: argilla
-      - name: Get date for conda cache
-        id: get-date
-        run: echo "::set-output name=today::$(/bin/date -u '+%Y%m%d')"
-        shell: bash
-      - name: Cache Conda env
-        uses: actions/cache@v3
+          python-version: "3.9"
+
+      - uses: actions/cache@v3
         id: cache
         with:
-          path: ${{ env.CONDA }}/envs
-          key: conda-${{ runner.os }}--${{ runner.arch }}--${{ steps.get-date.outputs.today }}-${{ hashFiles('environment_dev.yml') }}-${{ env.CACHE_NUMBER }}
-      - name: Update environment
+          path: ${{ env.pythonLocation }}
+          key: ${{ runner.os }}-python-${{ env.pythonLocation }}-${{ hashFiles('pyproject.toml') }}-test
+
+      - name: Install dependencies
         if: steps.cache.outputs.cache-hit != 'true'
-        run: mamba env update -n argilla -f environment_dev.yml
-      - name: Cache pip 👜
-        uses: actions/cache@v3
-        with:
-          path: ~/.cache/pip
-          key: ${{ runner.os }}-pip-${{ env.CACHE_NUMBER }}-${{ hashFiles('pyproject.toml') }}
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install -e .[server,postgresql,tests,integrations,listeners]
+          python -m spacy download en_core_web_sm
+
       - name: Set huggingface hub credentials
         if: github.ref == 'refs/heads/master' || startsWith(github.ref, 'refs/heads/releases')
         env:
           HF_HUB_ACCESS_TOKEN: ${{ secrets.HF_HUB_ACCESS_TOKEN }}
         run: echo "Enable HF access token"
+
       - name: Run tests 📈
         env:
           ARGILLA_ENABLE_TELEMETRY: 0
```

Comment:
I noticed we are not passing `HF_HUB_ACCESS_TOKEN` here but I know we do have some tests for this. Maybe it is a good moment to review this?